clang/test/CodeGenSYCL/check-direct-attribute-propagation.cpp

```diff
@@ -367,4 +367,4 @@ int main() {
 // CHECK: ![[NUM32]] = !{i32 16, i32 16, i32 32}
 // CHECK: ![[NUM88]] = !{i32 8, i32 8, i32 8}
 // CHECK: ![[NUM22]] = !{i32 2, i32 2, i32 2}
-// CHECK: ![[NUM123]] = !{i32 1, i32 2, i32 3}
+// CHECK: ![[NUM123]] = !{i32 3, i32 2, i32 1}
```

Comment:
We did. Similar to `reqd_work_group_size` the arguments should be reversed. That is, for the same arguments, the generated metadata should be the same for `reqd_work_group_size` and `work_group_size_hint`, which is now checked by the new clang/test/CodeGenSYCL/check-work-group-attributes-match.cpp test.

I suspect that the reason this hasn't been identified before is because this is a hint and as such is not guaranteed to have an effect, so users may have just assumed it wasn't taking effect on their systems.